examples/moving_horizon_estimation/mhe.py

```diff
@@ -0,0 +1,174 @@
+from mhe_simulation import run_simulation, check_results
+import biorbd
+import numpy as np
+import time
+
+from biorbd_optim import (
+    Instant,
+    OptimalControlProgram,
+    ProblemType,
+    Objective,
+    Constraint,
+    Bounds,
+    QAndQDotBounds,
+    InitialConditions,
+    ShowResult,
+    InterpolationType,
+    PlotType,
+    Data,
+)
+
+
+def warm_start_mhe(data_sol_prev):
+    q = data_sol_prev[0]["q"]
+    dq = data_sol_prev[0]["q_dot"]
+    u = data_sol_prev[1]["tau"]
+    x = np.vstack([q, dq])
+    X0 = np.hstack((x[:, 1:], np.tile(x[:, [-1]], 1)))  # discard oldest estimate of the window, duplicates youngest
+    U0 = u[:, 1:]  # discard oldest estimate of the window
+    X_out = x[:, 0]
+    return X0, U0, X_out
```

Comment:
Should we move this to an MHE module?

Reply:
Maybe when mature. I'm still working on improving the warm start...